Cannot name function as config when using #[get] macro

```rust
use actix_web::{App, HttpResponse, HttpServer, Responder, get};

#[get("/config")]
async fn config() -> impl Responder {
    HttpResponse::Ok().body("list of configurations")
}

#[actix_rt::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| {
        App::new()
            .service(config)
    })
    .bind("127.0.0.1:8088")?
    .run()
    .await
}

```
error
```
error[E0308]: mismatched types
 --> src\main.rs:3:1
  |
3 | #[get("/config")]
  | ^^^^^^^^^^^^^^^^^ expected struct `actix_web::config::AppService`, found struct `config`
  |
  = note: expected type `actix_web::config::AppService`
             found type `config`

error[E0308]: mismatched types
 --> src\main.rs:3:1
  |
3 | #[get("/config")]
  | ^^^^^^^^^^^^^^^^^ expected mutable reference, found fn item
  |
  = note: expected type `&mut actix_web::config::AppService`
             found type `fn() -> impl std::future::Future {<config as actix_web::service::HttpServiceFactory>::register::config}`

error: aborting due to 2 previous errors
```

--- a/actix-web-codegen/CHANGES.md
+++ b/actix-web-codegen/CHANGES.md
@@ -1,5 +1,9 @@
 # Changes
 
+## [0.2.NEXT] - 2020-xx-xx
+
+* Allow the handler function to be named as `config` #1290
+
 ## [0.2.0] - 2019-12-13
 
 * Generate code for actix-web 2.0

--- a/actix-web-codegen/src/route.rs
+++ b/actix-web-codegen/src/route.rs
@@ -195,15 +195,15 @@ impl Route {
             pub struct #name;
 
             impl actix_web::dev::HttpServiceFactory for #name {
-                fn register(self, config: &mut actix_web::dev::AppService) {
+                fn register(self, __config: &mut actix_web::dev::AppService) {
                     #ast
-                    let resource = actix_web::Resource::new(#path)
+                    let __resource = actix_web::Resource::new(#path)
                         .name(#resource_name)
                         .guard(actix_web::guard::#guard())
                         #(.guard(actix_web::guard::fn_guard(#extra_guards)))*
                         .#resource_type(#name);
 
-                    actix_web::dev::HttpServiceFactory::register(resource, config)
+                    actix_web::dev::HttpServiceFactory::register(__resource, __config)
                 }
             }
         };

--- a/actix-web-codegen/tests/test_macro.rs
+++ b/actix-web-codegen/tests/test_macro.rs
@@ -2,6 +2,12 @@ use actix_web::{http, test, web::Path, App, HttpResponse, Responder};
 use actix_web_codegen::{connect, delete, get, head, options, patch, post, put, trace};
 use futures::{future, Future};
 
+// Make sure that we can name function as 'config'
+#[get("/config")]
+async fn config() -> impl Responder {
+    HttpResponse::Ok()
+}
+
 #[get("/test")]
 async fn test_handler() -> impl Responder {
     HttpResponse::Ok()

